sockeye/inference.py

```diff
@@ -1339,10 +1362,12 @@ def _beam_search(self,
         # General data structure: each row has batch_size * beam blocks for the 1st sentence, with a full beam,
         # then the next block for the 2nd sentence and so on
 
+        best_word_indices = mx.nd.full((self.batch_size * self.beam_size,), val=self.start_id, ctx=self.context,
+                                       dtype='int32')
         # sequences: (batch_size * beam_size, output_length), pre-filled with <s> symbols on index 0
         sequences = mx.nd.full((self.batch_size * self.beam_size, max_output_length), val=C.PAD_ID, ctx=self.context,
                                dtype='int32')
-        sequences[:, 0] = self.start_id
+        sequences[:, 0] = best_word_indices
```

Comment:
is this faster, or why did you change this?

Reply:
yes, as said in the PR text it seems that indexing is slow. This actually gave a small speedup.

Reply:
actually, scratch that. I misread the position you were commenting on.
I can revert this.

Reply:
I mean it's fine, I was just wondering.